### ISSUE TITLE
[ItlySDK] - "Identify" event not fired for plugins (eg SegmentPlugin)

### ISSUE THREAD

**Issue by universe173** [amplitude/itly-sdk-ios#7]
### Issue
Please pass **userId** argument to **safeIdentify** and **safePostIdentify** here [Itly.swift#L61](https://github.com/amplitude/itly-sdk-ios/blob/235cf60aa24c1a749b8055777d31f5c2c23cc360/Sdk/Sdk/Itly.swift#L61) for firing **Identify** event for plugins.
You can see condition here [ItlySegmentPlugin.swift#L60](https://github.com/amplitude/itly-sdk-ios/blob/235cf60aa24c1a749b8055777d31f5c2c23cc360/SegmentPlugin/SegmentPlugin/ItlySegmentPlugin.swift#L60).

### Code:
Itly.swift

```
@objc public func identify(_ userId: String?, properties: Properties?) {
       guard !isDisabled else { return }

       validateAndRunOnAllPlugins(
            op: "identify",
            event: Event(name: "identify", properties: properties),
            includeContext: false,
            method: { (p, e) in
                try p.safeIdentify(nil, properties: e)
            },
            postMethod: { (p, e, validation) in
                try p.safePostIdentify(nil, properties: e, validationResults: validation)
            })
}
```

ItlySegmentPlugin.swift
```
public override func identify(_ userId: String?, properties: Properties?) {
        super.identify(userId, properties: properties)

        logger?.debug("\(self.id) identify(userId=\(userId ?? ""), properties=\(properties?.properties ?? [:]))")
        guard let userId = userId else {
            return
        }

        segmentClient?.identify(userId, traits: properties?.properties)
    }
```

**Comment by nyarian** [comment]
So, yeah, `identify` method does not seem to work, ever.

It seems that indeed `nil` is being passed to the `safeIdentify` (which is just a try-catch wrapper around `Plugin#identify`) [instead of the `userId` argument](https://github.com/amplitude/itly-sdk-ios/blob/master/Sdk/Sdk/Itly.swift#L61), and the plugin just [returns](https://github.com/amplitude/itly-sdk-ios/blob/master/SegmentPlugin/SegmentPlugin/ItlySegmentPlugin.swift#L61) if `userId` argument is `nil`.

Currently we are forced to call `identify` via Segment APIs to make it work, but we are loosing all the sense from Itly's schema there, as we need to support the `traits` protocol manually.

**Comment by justin-fiedler** [comment]
Hi @universe173 and @nyarian thanks for locating this error and bringing it to our attention. I made a PR that I believe resolves the issue. It you have a chance please review. I will try to get this merged and into the next release.

**Comment by justin-fiedler** [comment]
@universe173 and @nyarian I just released v1.0.3 of the SDK that includes this fix. Let me know how it goes. Thanks!
